src/include/brain/brain_defs.h

```diff
@@ -5,36 +5,136 @@ namespace terrier::brain {
 enum class ExecutionOperatingUnitType : uint32_t {
   INVALID,
 
+  /**
+   * AGGREGATE_BUILD
+   * num_rows: # input rows to aggregation
+   * cardinality: # unique values (although probably more correct is # of unique values based on group by clause)
+   */
   AGGREGATE_BUILD,
+  /**
+   * AGGREGATE_ITERATE
+   * num_rows: # rows output by aggregation  // TODO(WAN): when is this not 1?
+   * cardinality: # unique values output (either 1 for non-group by or # unique values based on group by)
+   */
   AGGREGATE_ITERATE,
 
+  /**
+   * HASHJOIN_BUILD
+   * num_rows: # input rows
+   * cardinality: # unique tuples
+   */
   HASHJOIN_BUILD,
+  /**
+   * HASHJOIN_PROBE
+   * num_rows: number of probes
+   * cardinality: number of matches
+   */
   HASHJOIN_PROBE,
 
+  /**
+   * IDXJOIN
+   * num_rows: N/A
+   * cardinality: N/A
+   * This feature doesn't actually exist.
+   * The recorder emits a IDX_SCAN (with num_loops = # rows output by the outer loop)
+   */
   IDXJOIN,
 
+  /**
+   * SORT_BUILD
+   * num_rows: # input tuples
+   * cardinality: # unique values
+   */
   SORT_BUILD,
+  /**
+   * SORT_ITERATE
+   * num_rows: # tuples output
+   * cardinality: # unique values
+   */
   SORT_ITERATE,
 
+  /**
+   * SEQ_SCAN
+   * num_rows: # tuples output (uncertain whether it's # accessed vs # after applying the filters)
+   * cardinality: # unique values // TODO(WAN): this should be the same right?
```

Comment:
In response to TODO, cardinality can probably be set to num_rows without anything going noticeably wrong (since a seq scan doesn’t really depend on table cardinality).

On a separate note, should probably amend the num_rows definition once a previous comment gets resolved?

Reply:
Done.